mmdet3d/models/roi_heads/bbox_heads/multi_group_head.py

```diff
@@ -0,0 +1,679 @@
+        reg_feat = self.feature_adapt_reg(x)
+
+        cls_score = self.cls_head(center_feat)
+        ret = self.task_head(reg_feat)
+        ret['hm'] = cls_score
+
+        return ret
+
+
+@HEADS.register_module
+class CenterHead(nn.Module):
+    """CenterHead for CenterPoint.
+
+    Args:
+        mode (str): Mode of the head. Default: '3d'.
+        in_channels (list[int] | int): Channels of the input feature map.
+            Default: [128].
+        tasks (list[dict]): Task information including class number
+            and class names. Default: [].
+        dataset (str): Name of the dataset. Default: 'nuscenes'.
+        weight (float): Weight for location loss. Default: 0.25.
+        code_weights (list[int]): Code weights for location loss. Default: [].
+        common_heads (dict): Conv information for common heads.
+            Default: dict().
+        crit (dict): Config of classification loss function.
+            Default: dict(type='CenterPointFocalLoss').
+        crit_reg (dict): Config of regression loss function.
+            Default: dict(type='L1Loss', reduction='none').
+        init_bias (float): Initial bias. Default: -2.19.
+        share_conv_channel (int): Output channels for share_conv_layer.
+            Default: 64.
+        num_hm_conv (int): Number of conv layers for heatmap conv layer.
+            Default: 2.
+        dcn_head (bool): Whether to use dcn_head. Default: False.
+        conv_cfg (dict): Config of conv layer.
+            Default: dict(type='Conv2d')
+        norm_cfg (dict): Config of norm layer.
+            Default: dict(type='BN2d').
+    """
+
+    def __init__(
+            self,
+            mode='3d',
+            in_channels=[
+                128,
+            ],
+            tasks=[],
+            train_cfg=None,
+            dataset='nuscenes',
+            weight=0.25,
+            code_weights=[],
+            common_heads=dict(),
+            crit=dict(type='CenterPointFocalLoss'),
+            crit_reg=dict(type='L1Loss', reduction='none'),
+            init_bias=-2.19,
+            share_conv_channel=64,
+            num_hm_conv=2,
+            dcn_head=False,
+            conv_cfg=dict(type='Conv2d'),
+            norm_cfg=dict(type='BN2d'),
+    ):
+        super(CenterHead, self).__init__()
+
+        num_classes = [len(t['class_names']) for t in tasks]
+        self.class_names = [t['class_names'] for t in tasks]
+        self.code_weights = code_weights
+        self.weight = weight  # weight between hm loss and loc loss
+        self.dataset = dataset
+        self.train_cfg = train_cfg
+
+        self.encode_background_as_zeros = True
+        self.use_sigmoid_score = True
+        self.in_channels = in_channels
+        self.num_classes = num_classes
+
+        self.crit = build_loss(crit)
+        self.crit_reg = build_loss(crit_reg)
+        self.loss_aux = None
+
+        self.box_n_dim = 9  # change this if your box is different
```

Comment:
Can be set according to the input argument dataset? like, if dataset == 'nuscenes': self.box_n_dim = 9; else NotImplementedError